api/http/Middleware.go

```diff
@@ -0,0 +1,19 @@
+package http
+
+import "net/http"
+
+type middleware func(http.Handler) http.Handler
+type chain []middleware
+
+func createChain(middlewares ...middleware) chain {
+	var slice chain
```

Comment:
We have middleware interface like: https://github.com/gorilla/mux/blob/master/middleware.go
where we can plug middleware like `.use(middleware)`

Plus also we can

`Next(value http.Handler) *HandlerBuilder {
	b.next = value
	return b
}`
Which help us plug the middleware in order.

For now we can go ahead with chaining approach and later improve it

Reply:
+1 to iterate later, I'd heavily hesitate to bring something from gorilla/mux in because gorilla/mux is unmaintained.

Reply:
I am not suggestion introducing gorilla mux, but implementing the middleware interface inspired by them